File: Easy_Level_using_Cpp/14. Longest Common Prefix.cpp
// Problem: 14. Longest Common Prefix
// Link: https://leetcode.com/problems/longest-common-prefix/description/
// Time Complexity: O(n.m)
// Space Complexity: O(1)
// Topics: String, Trie
// Solution:
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        if (strs.empty()) return "";
        for (int i = 0; i < strs[0].size(); i++) {
            char c = strs[0][i];
            for (int j = 1; j < strs.size(); j++) {
                if (i >= strs[j].size() || strs[j][i] != c) {
                    return strs[0].substr(0, i);
                }
            }
        }
        return strs[0];
    }
};

File: Easy_Level_using_Python/14. Longest Common Prefix.py
# Problem: 14. Longest Common Prefix
# Link: https://leetcode.com/problems/longest-common-prefix/description/
# Time Complexity: O(n.m)
# Space Complexity: O(1)
# Topics: String, Trie
# Solution:
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if (len(strs) == 0):
            return ""
        for i in range(0, len(strs[0])):
            c = strs[0][i]
            for j in range(1, len(strs)):
                if(i >= len(strs[j]) or c != strs[j][i]):
                    return strs[0][:i]
        return strs[0]